plugins/htpasswd/src/utils.ts

```diff
@@ -48,42 +60,76 @@ export async function verifyPassword(passwd: string, hash: string): Promise<bool
       bcrypt.compare(passwd, hash, (error, result) => (error ? reject(error) : resolve(result)))
     );
   } else if (hash.indexOf('{PLAIN}') === 0) {
-    return passwd === hash.substr(7);
+    return passwd === hash.slice(7);
   } else if (hash.indexOf('{SHA}') === 0) {
     return (
       crypto
         .createHash('sha1')
         // https://nodejs.org/api/crypto.html#crypto_hash_update_data_inputencoding
         .update(passwd, 'utf8')
-        .digest('base64') === hash.substr(5)
+        .digest('base64') === hash.slice(5)
     );
   }
   // for backwards compatibility, first check md5 then check crypt3
   return md5(passwd, hash) === hash || crypt3(passwd, hash) === hash;
 }
 
+/**
+ * generateHtpasswdLine - generates line for htpasswd file.
+ * @param {string} user
+ * @param {string} passwd
+ * @param {HtpasswdHashConfig} hashConfig
+ * @returns {string}
+ */
+export function generateHtpasswdLine(
+  user: string,
+  passwd: string,
+  hashConfig: HtpasswdHashConfig
+): string {
+  let hash: string;
+
+  switch (hashConfig.algorithm) {
+    case HtpasswdHashAlgorithm.bcrypt:
+      hash = bcrypt.hashSync(passwd, hashConfig.rounds);
```

Comment:
`generateHtpasswdLine` is used in asynchronous `changePasswordToHTPasswd`, so I think it would be great to use asynchronous `bcrypt.hash` instead of bcrypt.hashSync to prevent blocking.

I can try to do this in a separate PR :)

Reply:
Does `bcrypt.hash` worked out last time? I barely remember there were few issues with it. Doens't?

Reply:
I tried, and it seemed to work according to tests. Please, take a look here:
#589